monty_profiling/frameworks/models/evidence_matching.py

```diff
@@ -0,0 +1,316 @@
+        tolerances,
+        feature_weights,
+        feature_evidence_increment=1,
+        max_nneighbors=3,
+        initial_possible_poses="informed",
+        evidence_update_threshold="all",
+        vote_evidence_threshold=0.8,
+        past_weight=1,
+        present_weight=1,
+        vote_weight=1,
+        object_evidence_threshold=1,
+        x_percent_threshold=10,
+        path_similarity_threshold=0.1,
+        pose_similarity_threshold=0.35,
+        required_symmetry_evidence=5,
+        graph_delta_thresholds=None,
+        max_graph_size=0.3,
+        max_nodes_per_graph=2000,
+        num_model_voxels_per_dim=50,
+        use_multithreading=True,
+        gsg_class=FlopCountingEvidenceGoalStateGenerator,
+        gsg_args=None,
+        *args,
+        **kwargs,
+    ):
+        # Call parent's __init__ with all parameters
+        super().__init__(
+            max_match_distance=max_match_distance,
+            tolerances=tolerances,
+            feature_weights=feature_weights,
+            feature_evidence_increment=feature_evidence_increment,
+            max_nneighbors=max_nneighbors,
+            initial_possible_poses=initial_possible_poses,
+            evidence_update_threshold=evidence_update_threshold,
+            vote_evidence_threshold=vote_evidence_threshold,
+            past_weight=past_weight,
+            present_weight=present_weight,
+            vote_weight=vote_weight,
+            object_evidence_threshold=object_evidence_threshold,
+            x_percent_threshold=x_percent_threshold,
+            path_similarity_threshold=path_similarity_threshold,
+            pose_similarity_threshold=pose_similarity_threshold,
+            required_symmetry_evidence=required_symmetry_evidence,
+            graph_delta_thresholds=graph_delta_thresholds,
+            max_graph_size=max_graph_size,
+            max_nodes_per_graph=max_nodes_per_graph,
+            num_model_voxels_per_dim=num_model_voxels_per_dim,
+            use_multithreading=use_multithreading,
+            gsg_class=gsg_class,
+            gsg_args=gsg_args,
+            *args,
+            **kwargs,
+        )
+        self.flop_counter = None
+
+    def _update_evidence_with_vote(self, state_votes, graph_id):
+        """Count FLOPs for KDTree operations during vote processing."""
+        # Extract vote information
+        graph_location_vote = np.zeros((len(state_votes), 3))
+        vote_evidences = np.zeros(len(state_votes))
+        for n, vote in enumerate(state_votes):
+            graph_location_vote[n] = vote.location
+            vote_evidences[n] = vote.confidence
+
+        # Execute KDTree query
+        vote_location_tree = KDTree(
+            graph_location_vote,
+            leafsize=40,
+        )
+        vote_nn = 3  # TODO: Make this a parameter?
+        if graph_location_vote.shape[0] < vote_nn:
+            vote_nn = graph_location_vote.shape[0]
+
+        # Account for FLOPs for KDTree Construction
```

Comment:
Do you have a reference for what this was based on that we can link to?

Reply:
That's perfect thanks, yeah I think it would be great to have a comment somewhere (could be in Readme) that for k-d tree, we assume `O(knln_2(n))` for building vs. -- for querying.